baseapp/baseapp.go

```diff
@@ -761,3 +768,31 @@ func (app *BaseApp) runMsgs(ctx sdk.Context, msgs []sdk.Msg, mode runTxMode) (*s
 func makeABCIData(msgResponses []*codectypes.Any) ([]byte, error) {
 	return proto.Marshal(&sdk.TxMsgData{MsgResponses: msgResponses})
 }
+
+func (app *BaseApp) generateFraudProof(storeKeyToSubstoreTraceBuf map[types.StoreKey]*bytes.Buffer) FraudProof {
+	var fraudProof FraudProof
+	fraudProof.stateWitness = make(map[string]StateWitness)
+	cms := app.cms.(*multi.Store)
+
+	for storeKey, subStoreTraceBuf := range storeKeyToSubstoreTraceBuf {
+		kvStore := cms.GetKVStore(storeKey)
```

Comment:
This could be written as `cms.GetKVStore(storeKey).(*tracekv.Store).GetAllKeysUsedInTrace(*subStoreTraceBuf)` 
since you don't use `kvStore` or `traceKv` anywhere else.

Reply:
Updated